lib/Optimizer/CodeGen/LowerToQIR.cpp

```diff
@@ -724,9 +724,19 @@ class MeasureRewrite : public ConvertOpToLLVMPattern<OP> {
           LLVM::LLVMPointerType::get(rewriter.getIntegerType(8)));
       appendName = true;
     } else {
-      // If no register name is supplied, make one up.
+      // If no register name is supplied, make one up. Zero pad the counter so
+      // that sequential measurements contain alphabetically sorted register
+      // names.
       static unsigned counter = 0u;
-      regName = rewriter.getStringAttr("r" + std::to_string(counter++));
+      char regNameCounter[16];
```

Comment:
I guess that's my question - why 00009 vs 9?

Reply:
The `test/NVQPP/integration/load_value.cpp` was failing on the 3rd iteration without this change because there were 4 qubits per kernel call. The first was r0-r3. That worked fine. The second was r4-r7, and that worked fine. The 3rd was `r8, r9, r10, r11`, and when constructing the synthetic global register from that, it put the qubits in the order `r10, r11, r8, r9`. That caused the validation checks to fail because the outputs were in the wrong order. This change ensures that they will be placed in the correct order.

Reply:
I see. This makes sense then. Should we also figure out how to reset the register names? Like should each kernel invocation start from r0?

Reply:
In general, I'm not sure there's any harm in having it _not_ reset. Part of me likes it not reset because then I can easily tell which iteration is running. However, the other part of me thinks it's best if the user specifies a result name rather than letting the compiler auto-generate one, in which case, my first point is kind of moot. If you'd like me to pursue this, I may need some help from @schweitzpgi deciding what the best method is to determine if we're in a new kernel or not is. I suspect there may be some subtleties I may not be aware of since it was done this way to begin with.

Reply:
I would say don't let it derail this PR. You might just add an Issue Tracker for you. It might be as easy as adding the counter as an attribute on the LowerToQIR pass and passing it by reference to the Measure RewritePattern. That way for each lowering, you'd be starting from zero.

Reply:
> I would say don't let it derail this PR. You might just add an Issue Tracker for you. It might be as easy as adding the counter as an attribute on the LowerToQIR pass and passing it by reference to the Measure RewritePattern. That way for each lowering, you'd be starting from zero.

Sounds good.

Reply:
OK, I believe 82827a8f (new update for this PR) is doing what you asked for. I verified that it resets back to 0 on every iteration of the `load_value.cpp` test.